spec/import_spec.rb

```diff
@@ -48,35 +46,88 @@ def upload_file!(name, ext='csv')
     before do
       Author.create!(name: "John", last_name: "Doe")
       Author.create!(name: "Jane", last_name: "Roe")
-      add_post_resource({
-        validate: true,
-        headers_rewrites: {
-          :'Author Name' => :author_id
-        },
-        before_batch_import: ->(importer) do
-          authors_names = importer.values_at(:author_id)
-          # replacing author name with author id
-          authors   = Author.where(name: authors_names).pluck(:name, :id)
-          #{"Jane" => 2, "John" => 1}
-          options = Hash[*authors.flatten]
-          importer.batch_replace(:author_id, options)
-        end
-      })
-      visit "/admin/posts/import"
-      upload_file!(:posts)
     end
 
-    it "should resolve author_id by author name" do
-      Post.all.each do |post|
-        expect(Author.where(id: post.author.id)).to be_present
+    context "for csv for particular author" do
+      let(:author) { Author.take }
+
+      shared_examples 'successful inserts for author' do
+        it "should use predefined author_id" do
+          expect(Post.where(author_id: author.id).count).to eq(Post.count)
+        end
+
+        it "should be imported" do
+          expect(Post.count).to eq(2)
+          expect(page).to have_content "Successfully imported 2 posts"
+        end
+      end
+
+      context "no headers" do
+        before do
+          add_post_resource(
+              template_object: ActiveAdminImport::Model.new(author_id: author.id, csv_headers: [:title, :body, :author_id]),
```

Comment:
Line is too long. [124/80]<br>Indent the first parameter one step more than the previous line.